### DISCUSSION TITLE
Add support for additional analytics engines

### DISCUSSION THREAD

**Discussion by IgnusG** [featurehub-io/featurehub#324] [Q&A] [answered]
Currently only google analytics is available as a connector. Is it possible to support additional analytics engines as well or is that not considered for future roadmap plans?

**Describe the solution you'd like**
Probably a plugin-based connector approach with some core integrations with the big ones (GA etc.) and others through community-driven development. 

**Describe alternatives you've considered**
None that I'm aware of. 

**Additional context**
I'm mostly interested in a Posthog Analytics connector since its own feature flag implementation currently only supports boolean flags.

**Comment by rvowles** [comment]
Hiya @IgnusG - the Analytics stuff in the SDK is an interface, the Google Analytics is just one of the possible implementations. Which language are you using and I can point you to the things worth looking at.

**Comment by IgnusG** [comment]
Hey @rvowles. Thanks for the info about that the interfaces! I'm using JS/TS for my project.

**Comment by rvowles** [comment, marked as answer]
Hey there - its the AnalyticsCollector interface here: https://github.com/featurehub-io/featurehub/blob/master/sdks/typescript/client-typescript-core/app/analytics.ts

at the top - if you implement a class that does that, you can just register it like the Google one and normal the normal analytics interface works. The upcoming functionality doesn't change this, anything you do will work straight up.

**Comment by rvowles** [comment]
Going to close this issue unless there is something else?

**Comment by IgnusG** [comment]
That's all! Thanks @rvowles :)

> If I get a working prototype for posthog I might link back to it from here for other folks. Hope that's alright

**Comment by rvowles** [comment]
We would really appreciate that!

On Fri, Apr 16, 2021 at 8:02 PM Jonathan ***@***.***> wrote:

> That's all! Thanks @rvowles <https://github.com/rvowles> :)
>
> If I get a working prototype for posthog I might link back to it from here
> for other folks. Hope that's alright
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/featurehub-io/featurehub/issues/318#issuecomment-820996535>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAANTWB6OXDHGQVKTU43VCTTI7VH3ANCNFSM42WMFV5Q>
> .
>


-- 
---
Richard Vowles,
Full stack - from Kubernetes, through Node & Java, Web and Mobile
development in Flutter - software developer for hire!

ph: +64275467747
